web/src/main/java/com/epam/esm/controller/GiftCertificateController.java

```diff
@@ -23,49 +26,57 @@
 @RequestMapping("/api/certificates")
 public class GiftCertificateController {
     private GiftCertificateService certificateService;
+    private HateoasProvider<GiftCertificateDto> hateoasProvider;
 
-    public GiftCertificateController(GiftCertificateService certificateService) {
+    public GiftCertificateController(GiftCertificateService certificateService,
+                HateoasProvider<GiftCertificateDto> hateoasProvider) {
         this.certificateService = certificateService;
+        this.hateoasProvider = hateoasProvider;
     }
 
     /**
      * Retrieve certificates according to specified parameters.
      * All parameters are optional, so if they are not present, all certificates will be retrieved.
      *
      * @param searchParamsDto {@link GiftCertificateSearchParamsDto} instance
-     * @return JSON {@link ResponseEntity} object that contains list of {@link GiftCertificateDto}
+     * @return JSON {@link ResponseEntity} object that contains list of {@link HateoasModel} objects
      */
     @GetMapping
-    public ResponseEntity<List<GiftCertificateDto>> getCertificates(
-                @ModelAttribute GiftCertificateSearchParamsDto searchParamsDto) {
-        List<GiftCertificateDto> certificates = certificateService.find(searchParamsDto);
-        return new ResponseEntity<>(certificates, OK);
+    public ResponseEntity<List<HateoasModel>> getCertificates(
+            @ModelAttribute GiftCertificateSearchParamsDto searchParamsDto,
+            @ModelAttribute PageContext pageContext
+    ) {
+        List<GiftCertificateDto> certificates = certificateService.find(searchParamsDto, pageContext);
+        List<HateoasModel> models = HateoasModel.build(hateoasProvider, certificates);
+        return new ResponseEntity<>(models, OK);
     }
 
     /**
      * Retrieve certificate by its unique id.
      *
      * @param id certificate id
      * @throws EntityNotFoundException in case when certificate with this id does not exist
-     * @return JSON {@link ResponseEntity} object that contains {@link GiftCertificateDto} object
+     * @return JSON {@link ResponseEntity} object that contains {@link HateoasModel} object
      */
     @GetMapping("/{id}")
-    public ResponseEntity<GiftCertificateDto> getCertificate(@PathVariable("id") long id) {
+    public ResponseEntity<HateoasModel> getCertificate(@PathVariable("id") long id) {
```

Comment:
Iw would be better if we had a parametrization here like `ResponseEntity<HateoasModel<GiftCertificate>>`

Reply:
Fixed.